libunwind/src/UnwindRegistersSave.S

```diff
@@ -305,9 +305,22 @@ DEFINE_LIBUNWIND_FUNCTION(__unw_getcontext)
   mflr  0
   std   0, PPC64_OFFS_SRR0(3) // store lr as ssr0
   PPC64_STR(1)
+  PPC64_STR(4)      // Save r4 first since it will be used for fixing r2.
+#if defined(_AIX)
+  // The TOC register (r2) was changed by the glue code if unw_getcontext
+  // is called from a different module. Save the original TOC register
+  // in the context if this is the case.
+  mflr  4
+  lwz   4, 0(4)     // Get the first instruction at the return address.
+  lis   0, 0xe841   // Is it reloading the TOC register "ld 2,40(1)"?
```

Comment:
Why is the 32- and 64-bit code different?  Can't you use the same comparison technique?

Reply:
The legacy compiler uses xoris/cmplwi for both 32-bit and 64-bit.

Reply:
Changed to use the same comparison technique for 32- and 64-bit as suggested, thanks!